0019-glossary/0019-glossary.md

```diff
@@ -0,0 +1,272 @@
+It also contains the destination address and destination amount, on which connectors will base their routing decisions -
+both in choosing the onward path, and in deciding whether their incoming amount was big enough, in comparison to the destination
+amount mentioned in the ILP packet, to even merit an [forward action](#forward-action) at all. See also [routing table](#routing-table)
+
+## Interledger Payment Request (IPR)
+In the Interledger Payment Request (IPR) protocol, the receiver generates the payment details and condition. The receiver does not share the secret used to generate the condition and fulfillment with the sender or anyone else, but the sender must ask the recipient to generate and share a condition before sending each payment. IPR is primarily useful for building non-repudiable application layer protocols, in which the sender's posession of the fulfillment proves to third parties that the sender has paid the receiver for a specific obligation.
+See [IL-RFC 11](../0011-interledger-payment-request/0011-interledger-payment-request.md).
+
+# Interledger Protocol (ILP)
+The core of the Interledger protocol suite, described in [IL-RFC 3](../0003-interledger-protocol/0003-interledger-protocol.md).
+Colloquially the whole Interledger stack is sometimes referred to as "ILP". Technically, however, the Interledger Protocol is only one layer in the
+[Interledger protocol stack](#interledger-protocol-stack).
+
+## Interledger protocol stack
+The stack consisting of the [Interledger layer](#interledger-layer),
+a choice of [transport layers](#transport-layer), and an [application layer](#application-layer).
+
+## Interledger Quoting Protocol (ILQP)
+The protocol that defines how senders request quotes from connectors to determine the source or destination amount for an Interledger payment. Quoting is optional and senders MAY cache quotes and send repeated payments through the same connector.
+See [IL-RFC 8](../0008-interledger-quoting-protocol/0008-interledger-quoting-protocol.md).
+
+## Intermediate (transfer, ledger, amount)
+Any transfer/ledger/amount that is neither directly adjacent to the sender, nor to the receiver.
+
+## Ledger
+Stateful systems that track the ownership of [assets](#asset). Ledgers contain buckets of assets known as [accounts](#account) and record [transfers](#transfer) between them.
+Each account has a [balance](#balance), which is the amount of the ledger's assets the account holds. Account balances may be positive or negative, representing assets or liabilities.
+
+## Liquidity
+Maximum amount which can be moved back and forth over a route.
+
+## Local transaction
+A transaction that contains just one [transfer](#transfer) and zero [hops](#hop), thus staying within the same ledger.
+
+## Outgoing (transfer, ledger, amount)
+Relative to a connector, the transfer/ledger/amount that is located inbetween the connector and the receiver, directly adjacent to the connector.
```

Comment:
Rather than describing this in terms of adjacency, I think outgoing transfers would be easier to understand as the transfers of assets from a sender or connector to the sender or another connector.